pkg/volume/glusterfs/glusterfs.go

```diff
@@ -969,6 +970,11 @@ func getClusterNodes(cli *gcli.Client, cluster string) (dynamicHostIps []string,
 			return nil, fmt.Errorf("failed to get host ipaddress: %v", err)
 		}
 		ipaddr := dstrings.Join(nodeInfo.NodeAddRequest.Hostnames.Storage, "")
+		// IP validates if a string is a valid IP address.
+		ip := net.ParseIP(ipaddr)
```

Comment:
Can we add a unit test for this?

Reply:
@zouyee it would be better to add a unit test for this as @msau42 suggested.

Reply:
We should add this in master too, and then cherry pick both prs